.github/workflows/tests.yml

```diff
@@ -0,0 +1,130 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Run tests / create artifacts
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, ready_for_review]
+    branches: [ master ]
+
+jobs:
+  test-build-deploy:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./cdk
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Setup nodejs
+      uses: actions/setup-node@v2
+      with:
+        node-version: 16
+    - name: Install aws-cdk
+      run: npm install -g aws-cdk@1.110.1
+    - name: Install dependencies
+      run: |
+        pip install -r requirements.txt
+        pip install awscli==1.20.8 build
+    - name: Lint with flake8/black/isort
+      run: |
+        export FILES=(*.py domino_cdk tests)
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 ${FILES[@]} --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings.
+        flake8 ${FILES[@]} --count --exit-zero --statistics
+        black ${FILES[@]} --check
+        isort ${FILES[@]} --check
+    - name: Test with pytest
+      run: |
+        coverage run -m pytest tests
+
+    - name: Coverage report
+      run: |
+        coverage report
+    - name: Create/lint default config
+      env:
+        AWS_ACCOUNT_ID: ${{ secrets.INFRA_ACCOUNT_ID }}
+        GITHUB_SHA: ${{ github.sha }}
+        REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
+        REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+        GCR_JSON_CREDS: ${{ secrets.GCR_JSON_CREDS }}
+        ACM_CERT_ARN: ${{ secrets.ACM_CERT_ARN }}
+        BASE_DOMAIN: ${{ secrets.BASE_DOMAIN }}
+      run: |
+        export NAME=cdk-deploy-$(echo $GITHUB_SHA | cut -c1-6)
```

Comment:
since I had to look it up for `echo TAG=${GITHUB_REF:10}` line, might as well use `${GITHUB_SHA:0:5}`?